bugwarrior/services/bz.py

```diff
@@ -139,10 +145,13 @@ def get_keyring_service(service_config):
 
     @classmethod
     def validate_config(cls, service_config, target):
-        req = ['username', 'password', 'base_uri']
-        for option in req:
-            if option not in service_config:
-                die("[%s] has no 'bugzilla.%s'" % (target, option))
+        keys = ['username', 'password', 'api_key', 'base_uri']
+        if 'base_uri' not in service_config:
+            die("[%s] has no 'bugzilla.base_uri'" % (target,))
+        if 'username' not in service_config:
+            die("[%s] has no 'bugzilla.username'" % (target,))
+        if not ('password' in service_config or 'api_key' in service_config):
+            die("[%s] has no 'bugzilla.password' or 'bugzilla.api_key'" % (target,))
```

Comment:
The `keys` variable appears to be unused. My preference would be to stick with the loop where it works since that makes modifications to the required configuration values easy:

```py
req = ['username', 'base_uri']
for option in req:
    if option not in service_config:
        die("[%s] has no 'bugzilla.%s'" % (target, option))

if not ('password' in service_config or 'api_key' in service_config):
    die("[%s] has no 'bugzilla.password' or 'bugzilla.api_key'" % (target,))
```